python/run-tests.py

```diff
@@ -152,65 +172,17 @@ def parse_opts():
     return opts
 
 
-def _check_dependencies(python_exec, modules_to_test):
-    if "COVERAGE_PROCESS_START" in os.environ:
-        # Make sure if coverage is installed.
-        try:
-            subprocess_check_output(
-                [python_exec, "-c", "import coverage"],
-                stderr=open(os.devnull, 'w'))
-        except:
-            print_red("Coverage is not installed in Python executable '%s' "
-                      "but 'COVERAGE_PROCESS_START' environment variable is set, "
-                      "exiting." % python_exec)
-            sys.exit(-1)
-
-    # If we should test 'pyspark-sql', it checks if PyArrow and Pandas are installed and
-    # explicitly prints out. See SPARK-23300.
-    if pyspark_sql in modules_to_test:
-        # TODO(HyukjinKwon): Relocate and deduplicate these version specifications.
-        minimum_pyarrow_version = '0.8.0'
```

Comment:
How are we checking dependencies now?

Reply:
We are now relaying on the existing checks in the tests. For example:

https://github.com/apache/spark/blob/ab7b961a4fe96ca02b8352d16b0fa80c972b67fc/python/pyspark/sql/tests.py#L63-L69

https://github.com/apache/spark/blob/ab7b961a4fe96ca02b8352d16b0fa80c972b67fc/python/pyspark/sql/tests.py#L3121-L3123

which prints out a skip message like:

```
    test_createDataFrame_does_not_modify_input (pyspark.sql.tests.ArrowTests) ... skipped 
'PyArrow >= 0.8.0 must be installed; however, it was not found.'
```

which I am capturing here with a regex pattern.

Reply:
Gotcha. Thanks for the explanation!